apps/framework-docs-v2/src/components/mdx/interactive/use-persisted-state.ts

```
@@ -212,6 +261,60 @@ export function usePersistedState<T>(
+ * Handles both cross-tab (storage event) and same-page (custom event) synchronization.
+ *
+ * @param storageKey - Full storage key to watch
+ * @param onValueChange - Callback when value changes from storage/events
+ * @param deps - Dependency array for useEffect
+ */
+export function useStorageSync<T>(
+  storageKey: string | undefined,
+  onValueChange: (value: T | null) => void,
+  deps: React.DependencyList = [],
+): void {
+  useEffect(() => {
```

Biome (2.3.14)

[error] This hook does not specify its dependency on onValueChange.

This dependency is being used here, but is not specified in the hook dependency list.

This dependency is being used here, but is not specified in the hook dependency list.

This dependency is being used here, but is not specified in the hook dependency list.

React relies on hook dependencies to determine when to re-compute Effects.
Failing to specify dependencies can result in Effects not updating correctly when state changes.
These "stale closures" are a common source of surprising bugs.
Unsafe fix: Add the missing dependency to the list.


(lint/correctness/useExhaustiveDependencies)